python/pyabacus/src/py_diago_dav_subspace.cpp

```diff
@@ -0,0 +1,88 @@
+#include <complex>
+#include <functional>
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+#include <pybind11/complex.h>
+#include <pybind11/numpy.h>
+
+#include "module_hsolver/diago_dav_subspace.h"
+#include "module_hsolver/kernels/math_kernel_op.h"
+#include "module_base/module_device/types.h"
+
+namespace py = pybind11;
+using namespace pybind11::literals;
+
+void bind_diago_dav_subspace(py::module& m)
+{
+    py::module hsolver = m.def_submodule("hsolver");
+
+    py::class_<hsolver::diag_comm_info>(hsolver, "diag_comm_info")
+        .def(py::init<const int, const int>(), "rank"_a, "nproc"_a)
+        .def_readonly("rank", &hsolver::diag_comm_info::rank)
+        .def_readonly("nproc", &hsolver::diag_comm_info::nproc);
```

Comment:
You add `rank` and `nproc` here and I think it corresponds to the MPI Rank. However, you need to call `mpi4py` to run MPI in pyabacus. See the [playground](https://github.com/jieli-matrix/pybind-mpi/tree/main) and try!

Reply:
I plan to add parallel processing capabilities in the next PR. I will review the playground for reference. Thank you!